lib/web-server.js

```diff
@@ -1,49 +1,43 @@
-var fs = require('graceful-fs')
-var http = require('http')
-var https = require('https')
-var path = require('path')
-var connect = require('connect')
-var Promise = require('bluebird')
-
-var common = require('./middleware/common')
-var runnerMiddleware = require('./middleware/runner')
-var stopperMiddleware = require('./middleware/stopper')
-var stripHostMiddleware = require('./middleware/strip_host')
-var karmaMiddleware = require('./middleware/karma')
-var sourceFilesMiddleware = require('./middleware/source_files')
-var proxyMiddleware = require('./middleware/proxy')
-
-var log = require('./logger').create('web-server')
-
-var createCustomHandler = function (customFileHandlers, /* config.basePath */ basePath) {
+'use strict'
+
+const fs = require('graceful-fs')
+const http = require('http')
+const https = require('https')
+const path = require('path')
+const connect = require('connect')
+const Promise = require('bluebird')
+
+const common = require('./middleware/common')
+const runnerMiddleware = require('./middleware/runner')
+const stopperMiddleware = require('./middleware/stopper')
+const stripHostMiddleware = require('./middleware/strip_host')
+const karmaMiddleware = require('./middleware/karma')
+const sourceFilesMiddleware = require('./middleware/source_files')
+const proxyMiddleware = require('./middleware/proxy')
+
+const log = require('./logger').create('web-server')
+
+function createCustomHandler (customFileHandlers, config) {
   return function (request, response, next) {
-    for (var i = 0; i < customFileHandlers.length; i++) {
-      if (customFileHandlers[i].urlRegex.test(request.url)) {
-        return customFileHandlers[i].handler(request, response, 'fake/static', 'fake/adapter',
-          basePath, 'fake/root')
-      }
-    }
-
-    return next()
+    const handler = customFileHandlers.find((handler) => handler.urlRegex.test(request.url))
+    return handler
+      ? handler.handler(request, response, 'fake/static', 'fake/adapter', config.basePath, 'fake/root')
+      : next()
   }
 }
 
-createCustomHandler.$inject = ['customFileHandlers', 'config.basePath']
```

Comment:
Where did the injection keys end up?

Reply:
I removed them. Dependency injection framework use autoAnnonate in that case (read function parameter names)
https://github.com/vojtajina/node-di/blob/master/lib/injector.js#L70

After consideration I think it would be better to use `$inject` - I'm gonna fix it now.

Reply:
BTW, are `customFileHandlers` used in some way ? I see they are initialized here:
https://github.com/karma-runner/karma/blob/011a90c5945ad8141bd9b8063e12e0410780957d/lib/server.js#L74-L75

as empty array, and never changed (also comment 'TODO: remove' exists). Maybe we can get rid of everything related with that ?

Reply:
Revert `$inject` - done